### PR TITLE
chore: validate the printed code (fixes #71)

### LINKED ISSUE #71: need to validate printer output
Need to validate printer output

1.  Normal parse

2. If error, print diagnostic

3. If no error, print

4. Parse printed output and output diagnostic if errors

### PR DESCRIPTION
**Pull request by aladdin-add**

### CODE REVIEW

**Comment by aladdin-add** [comment]
seems there are some issues with the parser/printer, can you take a look? @KFlash

**Review by KFlash — APPROVED** [review]

**Comment by KFlash** [comment]
I'm soon going to publish a huge change to Kataw that will change it all, and solve a 7y try/fail and never succeed about comments so I'm also validating locale now both through re-parse and the transformers and visitors.

**Comment by KFlash** [comment]
@aladdin-add the issues you experienced is soon to be solved.

File: test/runner/tob.ts [hunk 1]
```diff
@@ -54,6 +55,16 @@ export async function file2Tob(filename: string): Promise<Tob> {
   tob.$printed = diagnostics.length === 0 ? printSourceFile(cst, tob.printerOptions) : '';
   tob.$diagnostics = diagnostics2md(diagnostics);
 
+  // check the pirnted code
+  if (tob.$printed && check) {
+    const $$diagnostics: any[] = [];
+    const cb = function (...args: any[]) {
+      $$diagnostics.push(args);
+    };
+    const _printedCst = (tob.parserOptions.module ? parseModule : parseScript)(tob.$printed, tob.parserOptions, cb);
```
Comment (KFlash) [inline test/runner/tob.ts:64]: @aladdin-add  What is the purpose with this variable '_printedCst'  ?

File: test/runner/tob.ts [hunk 2]
```diff
@@ -54,6 +55,16 @@ export async function file2Tob(filename: string): Promise<Tob> {
   tob.$printed = diagnostics.length === 0 ? printSourceFile(cst, tob.printerOptions) : '';
   tob.$diagnostics = diagnostics2md(diagnostics);
 
+  // check the pirnted code
+  if (tob.$printed && check) {
+    const $$diagnostics: any[] = [];
+    const cb = function (...args: any[]) {
```
Comment (KFlash) [inline test/runner/tob.ts:61]: this 'cb' is not in use, but still an redeclaration of the 'cb' above

Comment (aladdin-add) [reply test/runner/tob.ts:61]: it was used when parsing(3rd param).

it's not so convenient. what do you think the API like:

```js
const {cst, diagnostics} = parse(input, opts);
```

Comment (KFlash) [reply test/runner/tob.ts:61]: looks better!

Comment (KFlash) [reply test/runner/tob.ts:61]: 你能打开 discord 吗？ 我们需要讨论一些大问题

**Comment by KFlash** [comment]
@aladdin-add I think your issues has been solved with https://github.com/kataw/kataw/pull/138

**Pull request merged**